src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -675,6 +675,16 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (aliasesOrIndices == null || aliasesOrIndices.length == 0) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new ElasticsearchIllegalArgumentException("Null or zero length argument for list of index or alias names "
```

Comment:
we throw `IndexMissingException` all the time when `allowNoIndices` is set to true. I don't think we should act differently here.